pkg/kv/import.go

```diff
@@ -43,25 +43,58 @@ func (e *SafeEncoder) Encode(v interface{}) error {
 	return nil
 }
 
-func worker(ctx context.Context, jobChan <-chan *Entry, store Store) error {
+func consumer(ctx context.Context, cancel context.CancelFunc, jobChan <-chan *Entry, store Store) error {
 	for e := range jobChan {
 		if len(e.PartitionKey) == 0 {
+			cancel()
 			return fmt.Errorf("bad entry partition key: %w", ErrInvalidFormat)
 		}
 		if len(e.Key) == 0 {
+			cancel()
 			return fmt.Errorf("bad entry key: %w", ErrInvalidFormat)
 		}
 		if e.Value == nil {
+			cancel()
 			return fmt.Errorf("bad entry value: %w", ErrInvalidFormat)
 		}
 		err := store.SetIf(ctx, e.PartitionKey, e.Key, e.Value, nil)
```

Comment:
Maybe we can check for `ctx.Done()` here too, prevent redundant DB writes when we now we already fail?